### ISSUE TITLE
HTML5 keyboard focus

### ISSUE THREAD

**Issue by tjmolinski** [leafo/itch.io#112]
Hi, I'm unable to receive keyboard focus in my HTML5 game. I'm using pixi.js and just plain javascript to create the games. I have to tab until the focus is set on my game but this is really hard to do or drop down the console and focus on the game drop DOM element. Is there a better way to do this?

I've tried to request focus on the DOM element within javascript but that doesn't seem to work. Does anyone know what could be done?

This is one of my uploads that doesn't receive key focus.
http://tjmolinski.itch.io/must-go-faster

**Comment by fasterthanlime** [comment]
This is not an itch-specific issue, as it seems many games have problems when embedded in iframes.

[This article](http://www.qilineggs.com/2013/06/publishing-html5-game-to-kongregate-and.html) may help you with your problem :)

**Comment by tjmolinski** [comment]
Yeah thank you!
Attaching this little bit of code to the window's mousedown worked for me.
`function handleMouseDown(evt) {  
     window.focus();   
     evt.preventDefault();  
     evt.stopPropagation();  
     evt.target.style.cursor = 'default';  
   }`

**Comment by bryanwillis7** [comment]
@tjmolinski I know this was a long time ago, but is there any way you could provide more detail on where exactly you put this code? I tried putting it in my index.HTML file in some <script> tags, and it didn't work. :c It really sucks that pixi just doesn't play nice with itch... this problem should really be taken more seriously

**Comment by tjmolinski** [comment]
It's hard to give you a clear direction with the information you provided but I did find my old git repos that have this code. I was pretty much just referencing this javascript file from my html page after importing the pixi.js library.

`<head>
  <script src="lib/pixi.js"></script>
  <script src="src/main.js"></script>
</head>
`

and here is a copy of the main.js file that uses the bit of code to get auto focus.
https://pastebin.com/gpR7UMC8

Good luck!

**Comment by fasterthanlime** [comment]
Anyone finding this thread might be interested in a better solution here: https://github.com/itchio/itch.io/issues/727#issuecomment-337490481 (there's a pixi.js configuration option for that).
